scripts/Inelastic/Direct/ReductionWrapper.py

```diff
@@ -443,7 +447,43 @@ def _run_pause(self, timeToWait=0):
             Pause(timeToWait)
 
     #
+    def _check_progress_log_run_completed(self,run_number_requested):
+        """ Method to verify experiment progress log file and check if the file to reduce
+            has been written.
+            Input:
+             run_number_requested -- the number expected to be in logged in the log file
+
+            Output:
+              returns: (True,run_number_written,'') if the run_number stored in the log is
+                      higher then the run number requested
+                      (False,run_number_written,'') if the stored number is lower then the requested
+
+            If progress log is nod defined or not available, the method returns True, last known run number
+            and additional text information indicating the reason for failure
+            so further checks are necessary to verify if actual file is indeed available
+        """
+        propman = self.reducer.prop_man
+        if len(propman.arhive_upload_log_file)==0 :
+            return (True,0,'log test disabled as no log file available')
+
+        mod_time = os.path.getmtime(propman.arhive_upload_log_file)
+        if self._lat_commit_log_modification_time == mod_time: # Still old data in archive
+            run_num = self._last_runnum_added_to_archive
+            return (run_num >= run_number_requested,run_num,'no new data have been added to archive')
+        self._lat_commit_log_modification_time = mod_time
+        try:
+            with open(propman.arhive_upload_log_file) as fh:
+                contents = fh.read()
+        except:
```

Comment:
I think here we can just catch `OSError`?

Catching everything also picks up things like `KeyboardInterrrupt` that you don't want to catch.

Reply:
I have carefully looked into `OSError` and it is insufficient type of error to catch as `IOError` is different and also can be risen here. The issue comes from the fact that the log file can be suddenly modified by archive committing process, so the access may be denied or file destroyed or something else. The parent `EnvironmentError` may be suitable but the parent includes `KeyboardInterrupt`.  I think its reasonable to keep the code as it is and keyboard interrupt will be caught in other parts of the code, as it goes into waiting mode, if file access failed. We can of course try to catch `OSError` and `IOError` in sequence, but I am not sure this list is inclusive, bearing in mind any issue occurring with the file.

Reply:
This is the most uncertain part of the code with any errors difficult to reproduce and understand, so I better add additional exception check to the results analysis.